packages/core/contracts/optimistic-oracle-v2/implementation/WhitelistOptimisticOracleV2.sol

```diff
@@ -0,0 +1,275 @@
+        uint256 timestamp,
+        bytes memory ancillaryData
+    ) external nonReentrant() onlyOwner() {
+        require(
+            _getState(requester, identifier, timestamp, ancillaryData) == State.Requested,
+            "setEventBased: Requested"
+        );
+        Request storage request = _getRequest(requester, identifier, timestamp, ancillaryData);
+        request.requestSettings.eventBased = true;
+        request.requestSettings.refundOnDispute = true;
+    }
+
+    /**
+     * @notice Sets which callbacks should be enabled for the request.
+     * @param requester sender of the initial price request.
+     * @param identifier price identifier to identify the existing request.
+     * @param timestamp timestamp to identify the existing request.
+     * @param ancillaryData ancillary data of the price being requested.
+     * @param callbackOnPriceProposed whether to enable the callback onPriceProposed.
+     * @param callbackOnPriceDisputed whether to enable the callback onPriceDisputed.
+     * @param callbackOnPriceSettled whether to enable the callback onPriceSettled.
+     */
+    function ownerSetCallbacks(
+        address requester,
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData,
+        bool callbackOnPriceProposed,
+        bool callbackOnPriceDisputed,
+        bool callbackOnPriceSettled
+    ) external nonReentrant() onlyOwner() {
+        require(
+            _getState(requester, identifier, timestamp, ancillaryData) == State.Requested,
+            "setCallbacks: Requested"
+        );
+        Request storage request = _getRequest(requester, identifier, timestamp, ancillaryData);
+        request.requestSettings.callbackOnPriceProposed = callbackOnPriceProposed;
+        request.requestSettings.callbackOnPriceDisputed = callbackOnPriceDisputed;
+        request.requestSettings.callbackOnPriceSettled = callbackOnPriceSettled;
+    }
+
+    /**
+     * @notice Sets the proposer whitelist for a request.
+     * @param requester sender of the initial price request.
+     * @param identifier price identifier to identify the existing request.
+     * @param timestamp timestamp to identify the existing request.
+     * @param ancillaryData ancillary data of the price being requested.
+     * @param whitelist address of the whitelist to set.
+     */
+    function ownerSetProposerWhitelist(
+        address requester,
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData,
+        address whitelist
+    ) external nonReentrant() onlyOwner() {
+        customProposerWhitelists[_getId(requester, identifier, timestamp, ancillaryData)] = AddressWhitelistInterface(
+            whitelist
+        );
+    }
+
+    /**
+     * @notice Sets the requester whitelist.
+     * @param whitelist address of the whitelist to set.
+     */
+    function ownerSetRequesterWhitelist(address whitelist) external nonReentrant() onlyOwner() {
```

Comment:
for the sake of completeness, shall we also have similar function for setting the default proposer whitelist?